### ISSUE TITLE
Re-organize source tree

### ISSUE THREAD

**Issue by modeco80** [computernewb/collab-vm-server#103]
Ditto. Idea I had was like:

```
include/
 <same dirs as src/ for public headers of each component>
src/
  core/ (basically where all the core stuff goes w/o dependency to server code)
    sockets/ (maybe not?)
    vm/ (maybe subdirs for each hv?)
  server/ (where server code goes)
  websockets/
  vendor/ (below applies here too?)
      guacamole/ (UNTIL I GET RID OF THE WICKED WITCH.)
```
Will probably be done in parallel with cmake retry.

**Comment by yellows111** [comment]
> src/core/sockets/

sockets? what would *that* imply exactly?

> src/core/vm/*

sounds like an interesting idea, may need to consider what our hypervisor targers are going to be (all of them isn't a safe assumption)

nothing else to comment on, think the rest is fine.

**Comment by modeco80** [comment]
src/core/sockets would be sorta like src/Sockets, although I'm probably going to be moving Hv specific sockets into their src/core/vm/<hv> dir

src/core/vm/<hv> will have hv-specific code, yes, however src/core/vm/ will have base interfaces still

**Comment by yellows111** [comment]
thinking I should diverge this into a another issue/discussion, but what if we had a few extendable classes for generic stuff that's probably used all over the place? having the hv-specfic code derive from a `BaseController` or such and then have like `VncController` and stuff (`AintController`/`AgentController`, `SpiceController`) for adding new HV support down the line?
Might sound multi-layered to do classes like this, so i'm leaving it as just a thought

**Comment by modeco80** [comment]
I think just BaseVMController would be decent enough (I'd be using a base interface for display/control things, so the derived HV implementation class could just initalize the BaseVMController with e.g: `std::make_unique<core::VncDisplay>()` or something)

I'm intending to also maybe make controller creation a separate factory function (i.e: when creating VM controllers, the server just calls a factory function that takes something ala VMSettings that spits out a `shared_ptr<core::BaseVMController>` or something)

This might be ABI (and API hell too, since I'll probably be touching stuff a lot) hell but it'd make plugins a lot easier to extend e.g. addl. hypervisors

Might need a generic "VM Monitor" interface in the server side code instead of the qemu specific code tho? that's a problem the actual server code will face as well

**Comment by yellows111** [comment]
> Might need a generic "VM Monitor" interface in the server side code instead of the qemu specific code tho? that's a problem the actual server code will face as well

Not like we can do much about the missing feature parity for a hypervisor management protocol between all major vendors.

QEMU-specific things can stay qemu?, about other hypervisors just say like "Monitor not available" in AWA?

Considering how we should go through this.

**Comment by modeco80** [comment]
@computernewb master should be the default branch again cause this is mucking up github autoclosing (Techinically the 1.2.11 branch is "ahead" of master and so github keeps wanting me to PR it)

Anyways closing since I'm am implemented it.
